Разметка всех страниц

Lis2244 opened the pull request:
Столкнулся с делемой как я должен указать "Хочу сюда" ссылкой или кнопкой. Думаю что скорее ссылкой. Очень сомневаюсь в том что я правильно сделал верхний блок с картинками и логотипом. По мэйну тоже не понял как лучше разметить блоки.. но я бы сделал так). Более менее спокоен за навигацию..

---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/34/user/2080969/tasks/4)

Lis2244 committed:
Пробую разметить header и main

Столкнулся с делемой как я должен указать "Хочу сюда" ссылкой или кнопкой. Думаю что скорее ссылкой. Очень сомневаюсь в том что я правильно сделал верхний блок с картинками и логотипом. По мэйну тоже не понял как лучше разметить блоки.. но я бы сделал так). Более менее спокоен за навигацию..

Lis2244 committed:
Размечаю header и main второй раз

Footer завтра сделаю, понимаю что лучше не затягивать но пока быстрее не могу...

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
     <title>HTML Academy: Седона</title>
   </head>
   <body>
+    <header class="page-header">
+      <a class="logo-link" href="https://htmlacademy.ru/">
+        <img src="#" alt="Тут будет логотип"> 
+      </a>
+      <nav class="navigation">
+        <ul class="navigation-list">
+          <li class="navigation-item">
+            <a class="navigation-link" href="about.html">Главная</a> 
+          </li>
+          <li class="navigation-item">
+            <a class="navigation-link" href="index.html">О Седоне</a> 
+          </li>
+          <li class="navigation-item">
+            <a class="navigation-link" href="catalog.html">Гостиницы</a> 
+          </li>
+        </ul>
+        <section class="user-interface">
+          <ul class="navigation-list user-interface">
+            <li class="navigation-item">
+              <a class="navigation-link" href="about.html">Поиск</a> 
+            </li>
+            <li class="navigation-item">
+              <a class="navigation-link" href="index.html">Избранное</a> 
+            </li>
+          </ul> 
+            <a class="button-hochu-suda" href="#">Хочу сюда!</a>
+        </section>
+      </nav>
+    </header>
```
В header лишним будет тег section. Нету надобности оборачивать им список дополнительно. 

```
    <header class="page-header">
      <a class="logo-link" href="https://htmlacademy.ru/">
        <img src="#" alt="Тут будет логотип"> 
      </a>
      <nav class="navigation">
        <ul class="navigation-list"> переименуй тут в **site-navigation-list** 
          <li class="navigation-item"> **site-navigation-item** 
            <a class="navigation-link" href="about.html">Главная</a>  **site-navigation-link** 
          </li>
          <li class="navigation-item">
            <a class="navigation-link" href="index.html">О Седоне</a> 
          </li>
          <li class="navigation-item">
            <a class="navigation-link" href="catalog.html">Гостиницы</a> 
          </li>
        </ul>
          <ul class="navigation-list user-interface"> переименуй тут в  **user-navigation-list** 
            <li class="navigation-item"> **user-navigation-list-item**
              <a class="navigation-link" href="about.html">Поиск</a> 
            </li>
            <li class="navigation-item"> **user-navigation-list-item**
              <a class="navigation-link" href="index.html">Избранное</a>  **user-navigation-link**
            </li>
          </ul> 
            <a class="button-hochu-suda" href="#">Хочу сюда!</a>
      </nav>
    </header>
```

`button-hochu-suda` - плохое название для класса т.к испрользует в себе русский транслит. Имена для css-классов лучше стараться давать максимально абстрактные. Для кнопок можно использовать 

```
button
button-primary
button-outline
button-success
button-error
```
и т.д

Аналогично для ссылок

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+      <div class="hero-block">
+        <section class="upper-block">
+          <h1 class="visually-hidden">Седона</h1>
+            <img class="upper-block-img-welcom" src="#" alt="тут будет welcom">
+            <img class="upper-block to the gorgeous" src="#" alt="тут будет надпись to the gorgeous">
+            <img class="upper-block sedona-and-signature" alt="Тут будет текст большими буква Седона и подпись ниже">
```
```
<img class="upper-block-img-welcom" src="#" alt="тут будет welcom">
<img class="upper-block to the gorgeous" src="#" alt="тут будет надпись to the gorgeous">
<img class="upper-block sedona-and-signature" alt="Тут будет текст большими буква Седона и подпись ниже">
```

Эти три картинки будет лучше обьединить между собой и  экспортнуть как одну. В разметке тогда останется один тег img

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
-
+    <main class="main-index">
+      <div class="hero-block">
```
`<div class="hero-block">` - этот див не нужен)

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
-
+    <main class="main-index">
+      <div class="hero-block">
+        <section class="upper-block">
```
`class="upper-block"` Плохое название для класса) upper-block может в процессе жизнедейтельности приложения переехать вниз и название класса будет нелогичноек. 
Назови `s-greetings`
s сокращенно от section

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+            <img class="upper-block to the gorgeous" src="#" alt="тут будет надпись to the gorgeous">
+            <img class="upper-block sedona-and-signature" alt="Тут будет текст большими буква Седона и подпись ниже">
+        </section>
+      <section class="block-sedona">
+      <h2 class="description-of-sedona">Седона — небольшой городок в Аризоне, заслуживающий большего!</h2>
+        <p class="description-index">Рассмотрим 5 причин, по которым Седона круче, чем Гранд-Каньон!</p>
+       </section>
```
`class="block-sedona"` в `class="s-sedona"`

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+              <img class="city-rest-img" src="#" alt="маленький синий бутерброд">
+              <span class="header-city-rest">Еда</span>
+              <p class="text-city-rest">Всегда заказывайте топовый фирменный бургер, вы не разочаруетесь!</p>
+            </li>
+            <li class="city-rest-home">
+              <img class="city-rest-img" src="#" alt="маленький синий подарок">
+              <span class="header-city-rest">Сувениры</span>
+              <p class="text-city-rest">Не только китайского, но и настоящего местного производства!</p>
+            </li>
+          </ul>
+         <li class="bridge">
+           <img class="bridge-img" src="#" alt="Красивый каменный мост как из мультика Король Лев">
+           <div class="bridge-block">
+            <h3 class="bridge-header">Там есть мост дьявола</h3>  
+            <span class="reasons-namber">— №2 —</span>
+            <p class="bridge-info">Да, по нему можно пройти! Если вы осмелитесь, разумеется</p>
+           </div>
+         </li>
+         <li class="city-rest-bridge">
+           <div class="city-rest-bridge-block">
+             <h3 class="citi-header">Небольшая площадь</h3>
+             <span class="reasons-namber">— №3 —</span>
+             <p class="city-info">Все достопримечательности находятся очень близко</p>
+           </div>
+         </li>
+         <li class="city-rest-bridge-gray">
+          <div class="city-rest-bridge-block">
+            <h3 class="citi-header">Красивая дорога</h3>
+            <span class="reasons-namber">— №4 —</span>
+            <p class="city-info">Ехать в Седону из Лас-Вегаса совсем не скучно!</p>
+          </div>
+        </li>
+        <li class="city-rest-bridge">
+          <div class="city-rest-bridge-block">
+            <h3 class="citi-header">Мало туристов</h3>
+            <span class="reasons-namber">— №5 —</span>
+            <p class="city-info">Большинство едет в Гранд Каньон и толпится там</p>
+          </div>
+        </li>
+      </ol>
```
Список по структуре разетил почти верно. Проблема в нейминге css-классов. 

В header-е для списков использовал префиксы -list. Для единообразия стоит придерживаться единого стиля в нейминге по всей разметке. 

`the-first-list` - также не очень хорошее название. Если этот список переедет куда-то в другую часть сайта он уже не будет first-list)

`the-first-list` Назови например как `features-list`
Элементы списка лишки тогда будут `features-item`

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+        </li>
+      </ol>
+    </section>
+    <section class="hooked">
+      <h3 class="header-hooked">Заинтересовались?</h3>
+      <p class="hooked-text">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+      <a class="search-hotel" href="#"><scan class="shrift">ПОИСК ГОСТИНИЦЫ</scan> В седоне</a>
+    </section>
```
`ПОИСК ГОСТИНИЦЫ` весь текст в разметке маленькими буквами) Капслок потом задастся через стилизацию

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+      <p class="hooked-text">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+      <a class="search-hotel" href="#"><scan class="shrift">ПОИСК ГОСТИНИЦЫ</scan> В седоне</a>
+    </section>
+    </main>
   </body>
```
Не хватает еще блоков 

"Подписатсья на рассылку"
"Footer"

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+    <section class="hooked">
+      <h3 class="header-hooked">Заинтересовались?</h3>
+      <p class="hooked-text">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+      <a class="search-hotel" href="#"><scan class="shrift">ПОИСК ГОСТИНИЦЫ</scan> В седоне</a>
```
class="search-hotel" в `link`

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+          </div>
+          <img class="image-city" src="#" alt="Изображение городка с флагами на фоне гор">
+        </li>
+          <ul class="city-rest">
```
`city-rest` в `city-rest-list`

ydergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+          <img class="image-city" src="#" alt="Изображение городка с флагами на фоне гор">
+        </li>
+          <ul class="city-rest">
+            <li class="city-rest-home">
+              <img class="city-rest-img" src="#" alt="маленький синий домик">
+              <span class="header-city-rest">Жильё</span>
+              <p class="text-city-rest">Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
+            </li>
```
class="city-rest-home" в `rest-item`
class="city-rest-home-white" в `class="rest-item rest-item-bright"`
class="city-rest-home" в `rest-item`

yury-dergachev commented on index.html:
```diff
@@ -5,8 +5,114 @@
+          <img class="image-city" src="#" alt="Изображение городка с флагами на фоне гор">
+        </li>
+          <ul class="city-rest">
+            <li class="city-rest-home">
+              <img class="city-rest-img" src="#" alt="маленький синий домик">
+              <span class="header-city-rest">Жильё</span>
+              <p class="text-city-rest">Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
+            </li>
+            <li class="city-rest-home-white">
+              <img class="city-rest-img" src="#" alt="маленький синий бутерброд">
+              <span class="header-city-rest">Еда</span>
+              <p class="text-city-rest">Всегда заказывайте топовый фирменный бургер, вы не разочаруетесь!</p>
+            </li>
+            <li class="city-rest-home">
+              <img class="city-rest-img" src="#" alt="маленький синий подарок">
+              <span class="header-city-rest">Сувениры</span>
+              <p class="text-city-rest">Не только китайского, но и настоящего местного производства!</p>
+            </li>
```
class="city-rest-home" в `rest-item`
class="city-rest-home-white" в `class="rest-item rest-item-bright"`
class="city-rest-home" в `rest-item`

yury-dergachev commented:
Нужно еще настроить форматирование для редактора, чтобы автоматом при сохранее документа выставлялось одинаковое количество отступов для тегов  вложенных  друг в друга

Lis2244 committed:
Добавляю Каталог

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+                <span class="visually-hidden">Показать плитку</span>
+            </a>
+            <ul class="hotel-card-list">
+                <li class="hotel-card-item">
+                    <a class="hotel-card-link" href="#">
+                        <img class="hotel-card-img" src="#" alt="Пещера">
+                        <h3 class="hotel-card-title">Amara Resort & Spa</h3>
+                    </a>
+                    <ul class="hotel-card-subtitle-list">
+                        <li class="hotel-card-subtitle-item">Гостиница</li>
+                        <li class="hotel-card-subtitle-item">
+                            <b class="hotel-card-price">От 4000 ₽</b>
+                        </li>
+                    </ul>
+                    <ul class="button-list">
+                        <li class="button-item">
+                            <a class="button button-details" href="#">Подробнее</a>
+                        </li>
+                        <li class="button-item">
+                            <a class="button button-favourites" href="#">В избранное</a> 
+                        </li>
+                    </ul>
+                    <ul class="rating-list">
+                        <li class="stars-item">
+                            <img class="hotel-card-stars" src="#" alt="Звезда">
+                            <img class="hotel-card-stars" src="#" alt="Звезда">
+                            <img class="hotel-card-stars" src="#" alt="Звезда">
+                            <img class="hotel-card-stars" src="#" alt="Звезда">
+                        </li>
+                        <li class="rating-item">
+                            <p class="hotel-rating-item">Рейтинг: 8,5</p>
+                        </li>
+                    </ul>
+                </li>
```
В карточке перебор со списками) Самим списком молжно разметить только набор карточек, а ее внутренности для списков уже не подойдут. Можно использовать div, p, span и т.д

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+                    <a class="pagination-link" href="#">10</a>
+                </li>
+            </ul>
+            <ul class="selector-pagination-list">
+                <li class="selector-pagination-item">
+                    <p class="pagination-item-text">Гостиниц на странице:</p>
+                </li>
+                <li class="select-pagination-item">
+                    <select class="pagination-number" aria-label="Количество страниц">
+                        <option value="one">1</option>
+                        <option value="two">2</option>
+                        <option value="three">3</option>
+                        <option value="four">4</option>
+                    </select>
+                </li>
+            </ul>
```
```
<ul class="selector-pagination-list">
                <li class="selector-pagination-item">
                    <p class="pagination-item-text">Гостиниц на странице:</p>
                </li>
                <li class="select-pagination-item">
                    <select class="pagination-number" aria-label="Количество страниц">
                        <option value="one">1</option>
                        <option value="two">2</option>
                        <option value="three">3</option>
                        <option value="four">4</option>
                    </select>
                </li>
</ul>
```

Тут также не подойдет список. Список это как правило набор однородных элементов по смыслу, структуре.

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+    </main>
+
+    <footer class="footer-index">
+        <section class="s-mailing">
+          <img class="s-mailing-background" src="#" alt="фон">
+          <h3 class="header-mailing">Подпишитесь на рассылку</h3>
+          <p class="text-mailing">Только полезная информация и никакого спама, честное бойскаутское!</p>
+          <form class="s-mailing-form" action="http://echo.htmlacademy.ru" method="post">
+            <label class="visually-hidden" for="email">Email</label>
+            <input class="field" placeholder="Ваш e-mail" type="email" name="s-mailing-email" id="email">
+            <input class="button-mailing" type="submit" value="Подписаться">
+          </form>
+        </section>
```
```
<section class="s-mailing">
          <img class="s-mailing-background" src="#" alt="фон">
          <h3 class="header-mailing">Подпишитесь на рассылку</h3>
          <p class="text-mailing">Только полезная информация и никакого спама, честное бойскаутское!</p>
          <form class="s-mailing-form" action="http://echo.htmlacademy.ru" method="post">
            <label class="visually-hidden" for="email">Email</label>
            <input class="field" placeholder="Ваш e-mail" type="email" name="s-mailing-email" id="email">
            <input class="button-mailing" type="submit" value="Подписаться">
          </form>
        </section>
```

Это независимый раздел. Может жить отдельно от фетура. Надо вынести из него)

yury-dergachev commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            </li>
+          </ul>
+          <h3 class="contacts-phone">+7 (812) 812-12-12</h3>
+          <img class="contacts-logo-img" src="#" alt="Мелкий логотип академии">
```
`<img class="contacts-logo-img" src="#" alt="Мелкий логотип академии">` Нужно обернуть дополнительно в ссылку

Lis2244 committed:
Вношу правки

К секциям добавил как и обговаривали приставку s, убрал site префикс оставил только pagination-list без префикса. Убрал списки из списков)
